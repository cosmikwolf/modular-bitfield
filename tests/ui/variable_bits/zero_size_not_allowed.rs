use modular_bitfield::prelude::*;

#[derive(Specifier)]
#[bits(0, 8)]  // Zero size not allowed
enum TestData {
    #[discriminant = 0]
    Empty,
    #[discriminant = 1]
    Small(u8),
}

fn main() {}