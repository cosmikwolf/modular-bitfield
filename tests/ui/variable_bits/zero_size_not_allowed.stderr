error: bits sizes must be greater than 0
 --> tests/ui/variable_bits/zero_size_not_allowed.rs:4:8
  |
4 | #[bits(0, 8)]  // Zero size not allowed
  |        ^
